terraform/modules/vpc_network/main.tf

```diff
@@ -17,35 +17,52 @@
 
 module "vpc" {
   source       = "terraform-google-modules/network/google"
-  version      = "~> 4.0"
+  version      = "~> 5.0"
   project_id   = var.project_id
   network_name = var.vpc_network
   routing_mode = "GLOBAL"
 
   subnets = [
     {
-      subnet_name               = "vpc-01-subnet-01"
-      subnet_ip                 = "10.0.0.0/16"
-      subnet_region             = "us-central1"
+      subnet_name               = var.vpc_subnetwork
+      subnet_ip                 = var.subnet_ip
+      subnet_region             = var.region
+      subnet_private_access     = "true"
       subnet_flow_logs          = "true"
       subnet_flow_logs_interval = "INTERVAL_10_MIN"
       subnet_flow_logs_sampling = 0.7
       subnet_flow_logs_metadata = "INCLUDE_ALL_METADATA"
-    }
+    },
   ]
 
   secondary_ranges = {
-    vpc-01-subnet-01 = [
-      {
-        range_name    = "secondary-pod-range-01"
-        ip_cidr_range = "10.1.0.0/16"
-      },
-      {
-        range_name    = "secondary-service-range-01"
-        ip_cidr_range = "10.2.0.0/16"
-      },
+    (var.vpc_subnetwork) = [
+      var.secondary_ranges_pods,
+      var.secondary_ranges_services,
     ]
   }
+
+  firewall_rules = [
```

Comment:
Can you add comment on why we need a firewall rules here?

Reply:
Will do